src/common/dateInput/DateInputLogic.tsx

```diff
@@ -0,0 +1,240 @@
+  min: 0,
+  // 1) Ensures that a number pad is shown on iOS instead of the regular
+  // keyboard
+  // 2) Blocks iOS devices from inputting characters other than numbers
+  pattern: '\\d*',
+};
+
+type DateObject = {
+  date?: number;
+  month?: number;
+  year?: number;
+};
+
+export interface InputComponentProps {
+  id: string;
+  name: string;
+  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
+  value?: string;
+  innerRef?: React.RefObject<HTMLInputElement>;
+  min: number;
+  pattern: string;
+  type: string;
+  isInvalid: boolean;
+  label: string;
+}
+
+export interface WrapperComponentProps {
+  children?: React.ReactNode;
+  isInvalid: boolean;
+}
+
+interface Props {
+  onChange: (date: Date) => void;
+  wrapper: React.FC<WrapperComponentProps>;
+  value: Date | null;
+  isInvalid: boolean;
+  dateInputId: string;
+  monthInputId: string;
+  yearInputId: string;
+  dateInputName: string;
+  monthInputName: string;
+  yearInputName: string;
+  dateComponent: React.FC<InputComponentProps>;
+  monthComponent: React.FC<InputComponentProps>;
+  yearComponent: React.FC<InputComponentProps>;
+  dateInputLabel: string;
+  monthInputLabel: string;
+  yearInputLabel: string;
+}
+
+function getDateComponents(date: Date | null): DateObject {
+  if (!date) {
+    return {};
+  }
+
+  const monthDate = date.getDate();
+  const month = date.getMonth();
+  const year = date.getFullYear();
+
+  return {
+    date: monthDate,
+    month,
+    year,
+  };
+}
+
+function makeDate(date: DateObject): Date | null {
+  if (date.year && date.month && date.date) {
+    return new Date(date.year, date.month, date.date, 0, 0, 0, 0);
+  }
+
+  return null;
+}
+
+function getCharacterCount(number: number): number {
+  return number.toString().length;
+}
+
+function sanitizeInput(content: unknown) {
+  return String(content)
+    .split('')
+    .filter(character => DATE_INPUT_ALLOWED_CHARACTERS.includes(character))
+    .join('');
+}
+
+function cutToLength(string: string, length: number) {
+  if (string.length <= length) {
+    return string;
+  }
+
+  return string.slice(0, length);
+}
+
+function parseNumberFromValue(value: string, maxCharacterCount = 2): number {
+  // Removes all other characters than numbers from the result. This is
+  // a last fallback which makes negative numbers positive and removes
+  // possible characters from the output. The behavior of
+  // input[type="number"] is not consistent between platforms.
+  const sanitized = sanitizeInput(value);
+  const maxSized = cutToLength(sanitized, maxCharacterCount);
+
+  return Number(maxSized);
+}
+
+function DateInputLogic({
+  onChange,
+  value,
+  isInvalid,
+  wrapper,
+  dateInputId,
+  monthInputId,
+  yearInputId,
+  dateInputName,
+  monthInputName,
+  yearInputName,
+  dateComponent,
+  monthComponent,
+  yearComponent,
+  dateInputLabel,
+  monthInputLabel,
+  yearInputLabel,
+}: Props) {
+  const currentDate = getDateComponents(value);
```

Comment:
Is currentDate the date provided by the user or today's date? Could be helpful to change the name of the variable to something like `submittedDate` `userProvidedDate` or similar.

Reply:
It's not necessarily neither. It's the currently selected date from the perspective of the state. Due to how a `Date` object can be created--you need to know the date, month and year--we need to have two representations of a date at the same time:
* The values that represents the inputs' states (cached date: A temporary saving mechanism for parts of dates, until we can build a complete date object based on them, and send that to the consumer)
* The actual date that is this component communicates to its consumers (current date: this is the ultimate truth)

In this implementation the "current" date is not used for anything else than initialisation--but it's still a rather important concept for the architecture of this component. This is because we have two states that we need to sync at the correct moments. It's more or less an antipattern, but in this case it's hard to avoid and this is mostly the approach native HTML inputs take as well--one internal value and one external value. For instance we would need to sync the `currentDate` value into the inputs' state if we used this component in a context where some other input controlled the same value as this input does.

But alas the naming here stands to be improved, I only provided the background so you can make a better judgment on an appropriate name. Maybe something along the lines of internal/external. Cached -> temporary etc.